pandas/core/ops.py

```diff
@@ -740,32 +711,27 @@ def wrapper(left, right, name=name, na_op=na_op):
 
         converted = _Op.get_op(left, right, name, na_op)
 
-        left, right = converted.left, converted.right
         lvalues, rvalues = converted.lvalues, converted.rvalues
         dtype = converted.dtype
         wrap_results = converted.wrap_results
         na_op = converted.na_op
 
         if isinstance(rvalues, ABCSeries):
-            name = _maybe_match_name(left, rvalues)
+            res_name = _maybe_match_name(left, rvalues)
             lvalues = getattr(lvalues, 'values', lvalues)
             rvalues = getattr(rvalues, 'values', rvalues)
             # _Op aligns left and right
         else:
-            name = left.name
+            res_name = left.name
             if (hasattr(lvalues, 'values') and
                     not isinstance(lvalues, pd.DatetimeIndex)):
                 lvalues = lvalues.values
 
         result = wrap_results(safe_na_op(lvalues, rvalues))
-        return construct_result(
-            left,
-            result,
-            index=left.index,
-            name=name,
-            dtype=dtype,
-        )
+        return construct_result(left, result,
+                                index=left.index, name=res_name, dtype=dtype)
 
+    wrapper.__name__ = name
```

Comment:
I believe this is hit in tests, yes? (the name of the wrapper)

Reply:
Right now:
```
>>> pd.Series.__sub__
<unbound method Series.wrapper>
```

Reply:
hmm yeah that's not right. can you add tests for this? (I believe we have a suite of tests for the stat methods, so prob just need to add these on).

Reply:
We recently made similar tests for Index comparison method names in tests.indexes.test_base, I don't see much else to that effect.  I'll put something together.

Reply:
https://github.com/pandas-dev/pandas/blob/master/pandas/tests/generic/test_generic.py#L507